pkg/controllers/dynakube/extension/tls/reconciler.go

```diff
@@ -36,14 +36,14 @@ func NewReconciler(clt client.Client, apiReader client.Reader, dk *dynakube.Dyna
 }
 
 func (r *reconciler) Reconcile(ctx context.Context) error {
-	if r.dk.ExtensionsNeedsSelfSignedTLS() {
-		return r.reconcileSelfSigned(ctx)
+	if !r.dk.IsExtensionsEnabled() || !r.dk.ExtensionsNeedsSelfSignedTLS() {
```

Comment:
if I remember correctly (based on De Morgan's Thereom) we can refactor it using:

```
func (r *reconciler) Reconcile(ctx context.Context) error {
	if r.dk.IsExtensionsEnabled() && r.dk.ExtensionsNeedsSelfSignedTLS() {
		return r.reconcileSelfSignedTLSSecret(ctx)
	}

	return r.deleteSelfSignedTLSSecret(ctx)
}
```
I tested and it seems it works

Reply:
It will be refactored next week anyways with the condition, so I wouldn't worry too much

Reply:
[Done](https://github.com/Dynatrace/dynatrace-operator/pull/3950/commits/4d7f36c21f4399238e40cf0fa34e99e71cc422de)